packages/lib/pod/src/podUtil.ts

```diff
@@ -425,14 +465,17 @@ export function podValueFromRawValue(rawValue: PODRawValue): PODValue {
       // Check for a valid prefix. This is required to distinguish between EdDSA
       // public keys and strings. If there is no (valid) prefix, we assume an
       // encoded string.
-      const separatorIndex = rawValue.search(/:/);
-      const prefix = rawValue.slice(0, separatorIndex);
-      const payload = rawValue.slice(separatorIndex + 1);
-      return prefix === EDDSA_PUBKEY_PREFIX && payload.match(PUBLIC_KEY_REGEX)
-        ? { type: "eddsa-pubkey", value: payload }
-        : prefix === "string"
-        ? { type: "string", value: payload }
-        : { type: "string", value: rawValue };
+      if (rawValue.match(POD_STRING_TYPE_REGEX) !== null) {
+        const separatorIndex = rawValue.search(/:/);
```

Comment:
If you add a match group (parens around the prefix and key-string), you could the parts you need out of the regex match above as result[1] and result[2].
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_backreferences